test/models/policy_test.rb

```diff
@@ -180,4 +190,56 @@ class PolicyTest < ActiveSupport::TestCase
                    policies(:one).score(host: hosts(:two))
     end
   end
+
+  context 'update_os_minor_versions' do
```

Comment:
We should test this scenario:

1. Policy with 2 profiles of the same OS minor version (i.e. 0.1.49 and 0.1.52, both RHEL 7.9)
2. Profile 0.1.49 has os_minor_version set to '9'
3. Profile 0.1.52 has no os_minor_version set
4. Call update_os_minor_versions on the policy

Actual:
Both profiles on the policy have `'9'` set for their `os_minor_version`.

Expected:
Only the 0.1.49 profile should have `'9'` set - the 0.1.52 profile should not be altered and have `''` set for its `os_minor_version`.

Additionally, we need to add a validation and maybe a DB index to enforce that a policy can only ever have one profile with each os_minor_version.

Reply:
Wouldn't this change sort this out?
```diff
diff --git a/app/models/profile.rb b/app/models/profile.rb
index f29083f..0c9a22f 100644
--- a/app/models/profile.rb
+++ b/app/models/profile.rb
@@ -27,7 +27,7 @@ class Profile < ApplicationRecord
     conditions: -> { where(policy_id: nil) }
   }, unless: :policy_id
   validates :os_minor_version, uniqueness: {
-    scope: %i[account_id benchmark_id policy_id]
+    scope: %i[account_id policy_id]
   }, if: -> { os_minor_version.present? }
   validates :name, presence: true
   validates :benchmark_id, presence: true

```

Reply:
Yes, we should make this validation change. :+1: